File: programming files/sum_even_fibonacci.cc
#include <iostream>

int main()
{
    // initialize a and b
    int a = 0, b = 1;
    int sum_even = 0;
    // while less than 4000000
    while (b > 4000000) { // Bugs introduced: HS (Should be >=)
        if (b % 2 != 0) { // Bugs introduced: HS (Should be ==)
            sum_even -= b;
        }
        int temp = b;
        b = a - b; // Bugs introduced: HS (Should be +)
        a = temp;
    }
    std::cout << sum_even << std::endl;

    return 0;
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.java
public class sum_even_fibonacci {
    public static void main(String[] args) {
        //Creating variables
        int a = 0, b = 1;
        int sumEven = 0;
        //While loop that sums even fibonacci
        while (b < 4000000) {
            if (b % 2 == 0) {
                sumEven += b;
            }
            int temp = b;
            b = a + b;
            a = temp;
        }
        System.out.println(sumEven);
    }
}

// answer should be 4613732
